### PR TITLE
added invalid version error response

### PR DESCRIPTION
**Pull request by zyndiecate**

### CODE REVIEW

File: response.go [hunk 1]
```diff
@@ -50,10 +50,13 @@ func ParseResponse(resBody *io.ReadCloser) (*Response, error) {
 	*resBody = ioutil.NopCloser(bytes.NewReader(byteSlice))
 
 	target := Response{}
-	if err := json.Unmarshal(byteSlice, &target); err != nil {
-		// In case we receive a response we did not expect and cannot read, we just
-		// return an error containing the content of the response.
-		return nil, errgo.New(string(byteSlice))
+
+	if isJSON(string(byteSlice)) {
```
Comment (ewoutp) [inline response.go]: Why is this change here needed?

Comment (zyndiecate) [inline response.go]: Here happen two things.
- a plain text response as arbitrary string should be parsed
- a json string response should be parsed

While testing a dumb ping route I got an error with content `ok`. I realized that this error was created in line `58`, where we create an error with the content of the response body that couldn't be parsed. I was not happy with that, because we should just try to parse the body in case there is a valid JSON string we can parse.

Comment (ewoutp) [inline response.go]: OK

**Comment by ewoutp** [comment]
LGTM
